xmlschema/validators/groups.py

```diff
@@ -471,17 +471,20 @@ def iter_decode_children(self, elem, index=0, validation='lax'):
                 matched_choice = False
                 obj = None
                 for item in self:
-                    for obj in item.iter_decode_children(elem, child_index, validation):
-                        if not isinstance(obj, XMLSchemaValidationError):
-                            if isinstance(obj, tuple):
-                                yield obj
-                                continue
-                            if child_index < obj:
-                                matched_choice = True
-                                child_index = obj
-                        break
-                    if matched_choice:
-                        break
+                    try:
+                        for obj in item.iter_decode_children(elem, child_index, validation):
```

Comment:
I'd really like to pass `validation='strict'` here and remove `isinstance(obj, XMLSchemaValidationError)` logic on the next line. However, it seems `obj` can be `XMLSchemaValidationError` even when validation is strict.